File: src/models/Position.js
var mongoose = require('mongoose');

var positionSchema = mongoose.Schema({
    symbol: {type: String, required: true},
    strategyUuid: {type: String, required: true},
    transactionType: {type: String, required: true},
    timestamp: {type: Number, required: true},
    price: {type: Number, required: true},
    investment: {type: Number, required: true},
    profitability: {type: Number, required: true},
    closePrice: {type: Number, required: true},
    expirationTimestamp: {type: Number, required: true},
    closeTimestamp: {type: Number, required: true},
    profitLoss: {type: Number, required: true}
});

module.exports = mongoose.connection.model('Position', positionSchema);

File: lib/positions/position.cpp
#include "positions/position.h"

Position::Position(std::string symbol, time_t timestamp, double price, double investment, double profitability, int expirationMinutes) {
    this->symbol = symbol;
    this->timestamp = timestamp;
    this->price = price;
    this->investment = investment;
    this->profitability = profitability;
    this->closePrice = 0.0;
    this->isOpen = true;

    // TODO
    this->expirationTimestamp = this->timestamp + (expirationMinutes * 1000 * 60);
}

std::string Position::getSymbol() {
    return this->symbol;
}

time_t Position::getTimestamp() {
    return this->timestamp;
}

double Position::getPrice() {
    return this->price;
}

double Position::getClosePrice() {
    return this->closePrice;
}

double Position::getInvestment() {
    return this->investment;
}

double Position::getProfitability() {
    return this->profitability;
}

time_t Position::getCloseTimestamp() {
    return this->closeTimestamp;
}

time_t Position::getExpirationTimestamp() {
    return this->expirationTimestamp;
}

bool Position::getIsOpen() {
    return this->isOpen;
}

bool Position::getHasExpired() {
    // TODO
    return this->timestamp >= this->expirationTimestamp;
}

void Position::close(double price, time_t timestamp) {
    this->isOpen = false;
    this->closePrice = price;
    this->closeTimestamp = timestamp;
}
